pkg/driver/dra_hooks.go

```diff
@@ -93,6 +97,28 @@ func (np *NetworkDriver) PrepareResourceClaims(ctx context.Context, claims []*re
 	return result, nil
 }
 
+func (np *NetworkDriver) getNetlinkForDevice(nlHandle *netlink.Handle, deviceName string) (netlink.Link, error) {
+	// 1. Try by interface name from cache
+	ifName, ok := np.netdb.GetInterfaceName(deviceName)
+	if ok {
+		link, err := nlHandle.LinkByName(ifName)
+		if err == nil {
+			klog.V(4).Infof("Found link for device %s by interface name %s", deviceName, ifName)
+			return link, nil
+		}
+		klog.V(4).Infof("Could not find link by name %s for device %s, falling back to permanent address. error: %v", ifName, deviceName, err)
+	}
+
+	// 2. Fallback to permanent address
+	permAddr := strings.ReplaceAll(strings.TrimPrefix(deviceName, "mac-"), "-", ":")
```

Comment:
I had weird cases on the past with devices that stored the same mac address, and if we use virtual interfaces it may happen too, edge cases but will be hard to debug ... why do we need to change this to mac now? did we have any problem related to the interface name?